traitcuration/traits/datasources/ols.py

```diff
@@ -45,3 +49,54 @@ def get_ontology_id(term_iri):
     if ontology_id == 'orphanet':
         ontology_id = 'ordo'
     return ontology_id
+
+
+def get_term_status(is_obsolete, ontology_id=None):
+    """
+    Takes the ontology_id of a term and a whether it is obsolete or not, and returns its calculated status.
+    'Obsolete' if the is_obsolete flag is true, 'Deleted' if no info was found in any ontology, 'Current' if its
+    ontology is EFO, and 'Needs Import' if info about the term was found in another ontology.
+    """
+    if ontology_id is None:
+        return Status.DELETED
+    if is_obsolete:
+        return Status.OBSOLETE
+    if ontology_id == 'efo':
+        return Status.CURRENT
+    return Status.NEEDS_IMPORT
+
+
+@transaction.atomic
+def ols_update():
+    """
+    Query OLS for all terms with 'current', 'awaiting_import' and 'needs_import' status and update their status
+    """
+    for term in OntologyTerm.objects.filter(status__in=[Status.CURRENT, Status.AWAITING_IMPORT, Status.NEEDS_IMPORT]):
+        logger.info(f"Querying OLS for {term}")
+        efo_response = make_ols_query(term.iri, 'efo')
+        term_ontology_id = get_ontology_id(term.iri)
+        parent_ontology_response = None
+        if term_ontology_id != 'efo':
+            parent_ontology_response = make_ols_query(term.iri, term_ontology_id)
+        term.status = update_status(term, efo_response, parent_ontology_response)
+        term.save()
+
+
+def update_status(term, efo_response, parent_ontology_response):
+    """
+    Get the existing term status of a term and the OLS query response in EFO and in its parent ontology if any, and
+    compute the new status.
+    """
+    term_ontology_response = efo_response if term.status == Status.CURRENT else parent_ontology_response
+    if term.status == Status.AWAITING_IMPORT and efo_response is not None:
+        logger.info(f"FOUND IN EFO: Term {term}")
+        return Status.CURRENT
+    else:
+        if term_ontology_response is None:
+            logger.info(f"FOUND DELETED: Term {term}")
+            return Status.DELETED
+        elif term_ontology_response['is_obsolete'] is True:
```

Comment:
If I'm reading this right, it looks like we only check one ontology. However, for example, for terms which were imported from MONDO (or any other ontology), we need to check _both_ the parent ontology and the EFO. If the term is obsolete in _either,_ it should lead to it being declared obsolete in the status

Reply:
I'm kind of confused, a MONDO term can be obsolete in MONDO but not in EFO? (excluding bugs of course). Aren't they supposed to mirror each other?

Reply:
Yeah, that's complicated. As you can see, in the example you reported above, the label is different in Orphanet and EFO for the same term, presumably because it was edited in EFO after import.

My best current understanding is that "importing" is more like "copying" the term from other ontology into EFO. The identifier, including the parent ontology's prefix, is maintained, but the contents can change. I think that even during regular import, they sometimes edit the title and description compared to how it was in the original ontology. And consequently, if something changes in the parent ontology (including the obsolete status), it will probably not be automatically reflected in EFO.

Or maybe they do try to keep them in sync, it is just not always possible due to bugs or complexity of it all. I don't know. I have to admit that even after working with ontologies for 1.5 years, I still do not now how _all_ of the import rules work. I guess at some point I will have to contact SPOT and to clarify this.

For now, let's adopt this approach (it's different from what I said in the starting comment of this thread):
* Treat EFO as the main deciding ontology for both the label and deciding the deleted/obsolete status
* Still always query the parent ontology and report (log) all cases where there is any discrepancy between EFO and the parent ontology

Does this sound all right to you?

Reply:
Sounds quite reasonable, yes